waku/persistence/postgres/postgres.go

```diff
@@ -35,13 +36,30 @@ func WithDB(dburl string, migrate bool) persistence.DBOption {
 	}
 }
 
+func executeVacuum(db *sql.DB, logger *zap.Logger) error {
+	logger.Info("starting PostgreSQL database vacuuming")
+	_, err := db.Exec("VACUUM")
+	if err != nil {
+		return err
+	}
+	logger.Info("finished PostgreSQL database vacuuming")
+	return nil
+}
+
 // NewDB connects to postgres DB in the specified path
-func NewDB(dburl string) (*sql.DB, func(*sql.DB) error, error) {
+func NewDB(dburl string, shouldVacuum bool, logger *zap.Logger) (*sql.DB, func(*sql.DB) error, error) {
 	db, err := sql.Open("pgx", dburl)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	if shouldVacuum {
+		err := executeVacuum(db, logger)
```

Comment:
Just wondering, since vacuuming of db can take time upto minutes. 
Should this be a bocking call or should we start this process in a go-routine and report the result back in async way? 
Do we have any test indicating how much time a vacuum takes for our store db?

Reply:
Reading how SQLite `VACUUM` documentation, it will prevent writes to be executed, so it should be a blocking call.
No tests, but this command works by creating a temporary copy of the db, so the time it will take depends on the amount of data stored in the DB.

For postgresql, I modified the PR to use `VACUUM FULL`, just so the behavior of the command is the same as SQLite, but I think we shouldnt expect people to use vacuum with PostgreSQL since it is possible to setup routine vacuuming in it https://www.postgresql.org/docs/current/routine-vacuuming.html#AUTOVACUUM

Reply:
Ok, maybe the caller can do an async status to the user then. Rather than us doing it. 
I am just concerned since this is an API call by the user, blocking it for soo long may not be a good idea.

But its ok for now, we can change it if we get user feedback.